### PR TITLE
Add delays to tooltip + Typing Actions

### LINKED ISSUE #4683: Add delays to tooltips
Every time I'm using PostHog I'm swamped with tooltips that make it hard to navigate:

https://user-images.githubusercontent.com/38760734/121522278-5108b000-c9cb-11eb-9792-13f085223375.mov

I'd love for these to have a delay (e.g. show tooltip after 1sec hovering). Antd tooltips support `mouseEnterDelay`, but I guess the problem is then about how will the user know there's a tooltip at place X. Could be something to think about, and potentially add more info icons in different places?

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Fixes #4683

- [x] Replace all tooltips with custom tooltip component 
- [x] Add default 0.5s delay on mouse enter
- [x] Disable cursor events for the actual tooltip
- [x] Convert `<ActionStep/>` and `actionEditLogic` to Typescript
- [x] Change default tooltip placement to "top" 

The number of changed files for this PR is scary but don't be alarmed! This touches a lot of files only because I'm replacing all consumers of antd's `<Tooltip/>` with our own custom component. To review this PR, you can ignore all files not called `Tooltip.tsx`, `ActionStep.tsx`, and `actionEditLogic.ts` Ideally I'd want this PR to get merged in quickly to avoid any merge conflicts.

https://user-images.githubusercontent.com/13460330/129430356-ad5758b3-3b16-442c-9dee-18c4592f46a9.mov

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]
In general looks good and definitely an improvement!

The only thing I'd nit about is to also add types for the `props` for `actionEditLogic`. I see an `as any` to mask over missing types there.

File: frontend/src/lib/components/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { useState } from 'react'
+import { Tooltip as AntdTooltip } from 'antd'
+import { TooltipProps } from 'antd/lib/tooltip'
+import { useDebounce } from 'use-debounce'
+
+const DEFAULT_DELAY = 500 //ms
+
+// CAUTION: any changes here will affect tooltips across the entire app.
+export function Tooltip({ children, visible, ...props }: TooltipProps): JSX.Element {
```
Comment (paolodamico) [inline frontend/src/lib/components/Tooltip.tsx:9]: I think maybe we should add a prop to allow overriding this and keep the default behavior for specific places where it makes sense.

**Review by paolodamico — APPROVED** [review]

**Review by clarkus — APPROVED** [review]
Changes look good to me from a UX point of view. Thanks for working on this.

File: frontend/src/lib/components/Tooltip.tsx (hunk 1 above)
Comment (alexkim205) [reply frontend/src/lib/components/Tooltip.tsx:9]: got it I think that would help too. i'll add a `isDefault` prop here

**Pull request merged**